src/opts.c

```diff
@@ -597,6 +597,17 @@ natsOptions_SetMaxPingsOut(natsOptions *opts, int maxPignsOut)
     return NATS_OK;
 }
 
+natsStatus
+natsOptions_SetIOBufSize(natsOptions *opts, int ioBufSize)
+{
+  LOCK_AND_CHECK_OPTIONS(opts, (ioBufSize < 0));
+
+  opts->ioBufSize = ioBufSize;
```

Comment:
I think you need to check for ioBufSize being 0 here, and if so, use the default value. The default value is set when creating an option object, but if user calls this with SetIOBufSize(opts, 0), then I think that would set it to zero. (see comment about the test, which I think if you write first should fail).

Reply:
>  if user calls this with SetIOBufSize(opts, 0), then I think that would set it to zero.

This is no problem because when `opts->ioBufSize` is set to zero, the default value will be set in `natsConn_create`.
https://github.com/nats-io/nats.c/pull/250/files#diff-152747a2399765ea9211ffbfd6ba74edR2978

Some other options (e.g. maxPingsOut) are implemented with this pattern as well.

(I also feel the pattern is a bit potentially dangerous though.)

Reply:
By the way, I will fix the indent here.

Reply:
See 643cab1.